src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1173,14 +1173,17 @@ private void ParseModifiers(SyntaxListBuilder tokens, bool forAccessors)
                 {
                     case DeclarationModifiers.Partial:
                         var nextToken = PeekToken(1);
-                        var isPartialType = this.IsPartialType();
-                        var isPartialMember = this.IsPartialMember();
-                        if (isPartialType || isPartialMember)
+                        var isPartialType = CanFollowPartialKeywordOfPartialType(nextToken, out bool isPartialFollowedByType);
```

Comment:
This approach seems to only search forward one token, but I think we should try searching forward as many tokens as necessary to disambiguate. I would expect the code to look a lot like `IsPartialMember`.

Reply:
I think the variable name is misleading. it's true only when the next token can follow `partial` whether it's `struct`/`class` or another modifier.

Reply:
Also I think we should get rid the of the out parameter (since we need to discard it in other places), extracting that logic isn't too bad. Thoughts?